lib/puppet/functions/yaml_data.rb

```diff
@@ -21,7 +21,7 @@ def yaml_data(options, context)
     path = options['path']
     context.cached_file_data(path) do |content|
       begin
-        data = YAML.load(content, path)
+        data = Puppet::Util::Yaml.safe_load(content, [Symbol], path)
```

Comment:
Since this is a function, could someone be relying on the old behavior of returning arbitrary yaml data?

Reply:
I doubt it, since we didn't allow anything but `RichData` to be used by lookup (this is the lookup `yaml_data` function).